src/Adapter/Grid/Data/Factory/CartGridDataFactory.php

```diff
@@ -0,0 +1,156 @@
+     * @param LocaleInterface $locale
+     * @param ContextStateManager $contextStateManager
+     */
+    public function __construct(
+        GridDataFactoryInterface $doctrineDataFactory,
+        TranslatorInterface $translator,
+        LocaleInterface $locale,
+        ContextStateManager $contextStateManager
+    ) {
+        $this->doctrineDataFactory = $doctrineDataFactory;
+        $this->translator = $translator;
+        $this->locale = $locale;
+        $this->contextStateManager = $contextStateManager;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getData(SearchCriteriaInterface $searchCriteria)
+    {
+        $data = $this->doctrineDataFactory->getData($searchCriteria);
+
+        $records = $this->applyModifications($data->getRecords());
+
+        return new GridData($records, $data->getRecordsTotal(), $data->getQuery());
+    }
+
+    /**
+     * @param RecordCollectionInterface $records
+     *
+     * @return RecordCollection
+     *
+     * @throws CartException
+     */
+    private function applyModifications(RecordCollectionInterface $records)
+    {
+        $modifiedRecords = [];
+
+        foreach ($records as $record) {
+            switch ($record['status']) {
+                case 'abandoned_cart':
+                    $record['status'] = $this->translator->trans(
+                        'Abandoned cart',
+                        [],
+                        'Admin.Orderscustomers.Feature'
+                    );
+                    break;
+                case 'not_ordered':
+                    $record['status'] = $this->translator->trans('Not placed', [], 'Admin.Orderscustomers.Feature');
+                    break;
+            }
+
+            if (empty($record['carrier_name'])) {
+                $record['carrier_name'] = '--';
+            }
+
+            if (empty($record['customer_name'])) {
+                $record['customer_name'] = '--';
+            }
+
+            $record['online'] = $record['id_guest'] ?
+            $this->translator->trans('Yes', [], 'Admin.Global') :
+            $this->translator->trans('No', [], 'Admin.Global');
+
+            $cart = new Cart($record['id_cart']);
+            $this->contextStateManager->setLocale($this->locale);
+            $this->contextStateManager->setCart($cart);
+            $this->contextStateManager->setCurrency(new Currency((int) $cart->id_currency));
+            $this->contextStateManager->setCustomer(new Customer((int) $cart->id_customer));
+
+            $record['cart_total'] = Cart::getTotalCart($cart->id, true, Cart::BOTH_WITHOUT_SHIPPING);
+            $record['is_order_placed'] = $cart->orderExists();
+
+            if (!isset($cart->id_shop)) {
+                throw new CartException('cart shop id is not set');
+            }
+            $record['shop_name'] = Context::getContext()->shop->getShops()[$cart->id_shop]['name'];
+
+            $modifiedRecords[] = $record;
+        }
+
```

Comment:
I think we need to call `$this->contextStateManager->restoreContext();` when we're done processing this cart data

Reply:
added it before the return statement of the Controller action

Reply:
Thank you for this notice, will change it now